src/xrt/drivers/illixr/illixr_component.cpp

```diff
@@ -78,31 +81,44 @@ extern "C" void illixr_write_frame(unsigned int left,
 
 	illixr_plugin_obj->sb_eyebuffer.put(illixr_plugin_obj->sb_eyebuffer.allocate<rendered_frame>(
 	    rendered_frame {
-	        std::array<GLuint, 2>{ left, right }.data(),
-	        std::array<GLuint, 2>{ buffer_to_use, buffer_to_use }.data(),
+	        std::array<GLuint, 2>{ left, right },
+	        std::array<GLuint, 2>{ buffer_to_use, buffer_to_use }, // .data() deleted FIXME
             illixr_plugin_obj->prev_pose,
             illixr_plugin_obj->sample_time,
-            std::chrono::system_clock::now()
+			illixr_plugin_obj->rtc->now()
         }
     ));
 
     buffer_to_use = (buffer_to_use == 0U) ? 1U : 0U;
 }
 
+// extern "C" int64_t illixr_get_vsync_ns() {
+// 	assert(illixr_plugin_obj != nullptr && "illixr_plugin_obj must be initialized first.");
+
+//     switchboard::ptr<const switchboard::event_wrapper<time_point>> vsync_estimate = illixr_plugin_obj->sb_vsync_estimate.get_ro_nullable();
+	
+// 	// if (vsync_estimate == nullptr)
+// 	// {
+// 	// 	return std::chrono::duration_cast<std::chrono::nanoseconds>((illixr_plugin_obj->rtc->now()).time_since_epoch()).count() + NANO_SEC/60;
+// 	// }
+
+// 	// return std::chrono::duration_cast<std::chrono::nanoseconds>((**vsync_estimate).time_since_epoch()).count();
+// 	time_point target_time = vsync_estimate == nullptr ? illixr_plugin_obj->rtc->now() + freq2period(60.0f) : **vsync_estimate;
+
+// 	return illixr_plugin_obj->rtc->absolute_ns(target_time);
+// }
+
 extern "C" int64_t illixr_get_vsync_ns() {
 	assert(illixr_plugin_obj != nullptr && "illixr_plugin_obj must be initialized first.");
 
-    switchboard::ptr<const switchboard::event_wrapper<time_type>> vsync_estimate = illixr_plugin_obj->sb_vsync_estimate.get_ro_nullable();
+    switchboard::ptr<const switchboard::event_wrapper<time_point>> vsync_estimate = illixr_plugin_obj->sb_vsync_estimate.get_ro_nullable();
 	
-	if (vsync_estimate == nullptr)
-	{
-		return std::chrono::duration_cast<std::chrono::nanoseconds>((std::chrono::system_clock::now()).time_since_epoch()).count() + NANO_SEC/60;
-	}
+	time_point target_time = vsync_estimate == nullptr ? illixr_plugin_obj->rtc->now() + freq2period(60.0f) : **vsync_estimate;
```

Comment:
Do something like `static constexpr duration VSYNC_PERIOD {freq2period(60.0)};` at the top of the class and then reference `VSYNC_PERIOD` here.